pkg/controllers/workapplier/controller_test.go

```diff
@@ -0,0 +1,490 @@
+/*
```

Comment:
please make sure that the previous test and integration test still works

Reply:
Will do. They will be sent in a separate PR if that's OK (some minor adjustments need to be done -> almost completed, e.g., the original test suites does not distinguish between hub/member envs; and there are some new/updated conditions, etc.).